instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java

```diff
@@ -148,27 +148,49 @@ private String findByCurrentDirectoryApplicationProperties() {
     } catch (Exception e) {
       // expected to fail sometimes
     }
-    logger.log(Level.FINER, "Checking application.properties in current dir: {0}", result);
+    logger.log(FINER, "Checking application.properties in current dir: {0}", result);
     return result;
   }
 
+  @Nullable
+  private String findByClasspathApplicationYml() {
+    return findByClasspathYamlFile("application.yml");
+  }
+
   @Nullable
   private String findByClasspathApplicationYaml() {
-    String result =
-        loadFromClasspath("application.yml", SpringBootServiceNameDetector::parseNameFromYaml);
-    logger.log(Level.FINER, "Checking application.yml in classpath: {0}", result);
+    return findByClasspathYamlFile("application.yaml");
+  }
+
+  private String findByClasspathYamlFile(String fileName) {
+    String result = loadFromClasspath(fileName, SpringBootServiceNameDetector::parseNameFromYaml);
+    if (logger.isLoggable(FINER)) {
+      logger.log(FINER, "Checking {0} in classpath: {1}", new Object[] {fileName, result});
+    }
     return result;
   }
 
+  @Nullable
+  private String findByCurrentDirectoryApplicationYml() {
+    return findByCurrentDirectoryYamlFile("application.yml");
+  }
+
   @Nullable
   private String findByCurrentDirectoryApplicationYaml() {
+    return findByCurrentDirectoryYamlFile("application.yaml");
+  }
+
+  @Nullable
+  private String findByCurrentDirectoryYamlFile(String fileName) {
     String result = null;
-    try (InputStream in = system.openFile("application.yml")) {
+    try (InputStream in = system.openFile(fileName)) {
       result = parseNameFromYaml(in);
```

Comment:
I think there should be a null check here to avoid NPE

Reply:
This one loads the file from the file system, `Files.newInputStream()` never returns nulls